### PR TITLE
OAK-9140: Don't use stored index definition in elastic indexes

### PR DESCRIPTION
**Pull request by averma21**

### CODE REVIEW

**Review by chibulcuteanu — APPROVED** [review]

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java [hunk 1]
```diff
@@ -79,17 +79,14 @@
...
     public final int bulkRetries;
     public final long bulkRetriesBackoff;
-    private final String indexPrefix;
     private final String remoteAlias;
 
     private final Map<String, List<PropertyDefinition>> propertiesByName;
 
     public ElasticIndexDefinition(NodeState root, NodeState defn, String indexPath, String indexPrefix) {
-        super(root, getIndexDefinitionState(defn), determineIndexFormatVersion(defn), determineUniqueId(defn), indexPath);
-        boolean isReindex = defn.getBoolean(IndexConstants.REINDEX_PROPERTY_NAME);
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java:89]: `IndexConstants` not used anymore. Remove from imports

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java [hunk 2]
```diff
@@ -79,17 +79,14 @@
...
-        String indexSuffix = "-" + (getReindexCount() + (isReindex ? 1 : 0));
-        this.indexPrefix = indexPrefix != null ? indexPrefix : "";
-        this.remoteAlias = ElasticIndexNameHelper.getIndexAlias(indexPrefix, getIndexPath());
+        super(root, defn, determineIndexFormatVersion(defn), determineUniqueId(defn), indexPath);
+        String indexSuffix = "-" + getReindexCount();
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java:87]: should we start using the `seed` here?

**Review by fabriziofortino — COMMENTED** [review]
Just a couple of minor comments.

Moreover, would it be possible to include a unit test that captures the issue related to the use of stored index definitions?

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java (hunk 2 above)
Comment (thomasmueller) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java:87]: I'm not quite sure if the getReindexCount() now makes sense here... we will see! If tests start failing, then maybe it doesn't work any longer.

**Review by thomasmueller — APPROVED** [review]

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java (hunk 2 above)
Comment (thomasmueller) [reply oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java:87]: I would prefer seed as well!

**Comment by averma21** [comment]
@fabriziofortino 

> would it be possible to include a unit test that captures the issue related to the use of stored index definitions?

The issue was not because of using stored index index definitions, but rather due to improperly using it. So, I think that test case won't make sense. 

However for elastic index I can try to add a test case for checking that stored index definition is not created.

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java (hunk 2 above)
Comment (averma21) [reply oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java:87]: We still need some suffix to handle re-indexing transparently. As discussed we need to replace `reindexCount` with `seed` here.

Comment (averma21) [reply oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java:87]: Yes. I think the change would not be too small and would be unrelated, so I will create a new issue for that.

**Pull request closed without merging**